qa/rpc-tests/quality_voiding.py

```diff
@@ -0,0 +1,362 @@
+        assert_true(cert_2_epoch_0 in self.nodes[0].getblock(mined, True)['cert'])
+
+        quality = 110
+        amount_cert_3 = [{"pubkeyhash": pkh_node2, "amount": bwt_amount_3}]
+        mark_logs("Create Cert1 with quality {}, bwt transfer {} and place it in mempool".format(quality, bwt_amount_3), self.nodes, DEBUG_MODE)
+        proof = mcTest.create_test_proof(
+            "sc1", epoch_number, epoch_block_hash, prev_epoch_block_hash,
+            quality, constant, [pkh_node2], [bwt_amount_3])
+        try:
+            cert_3_epoch_0 = self.nodes[1].send_certificate(scid, epoch_number, quality, epoch_block_hash, proof,
+                                                            amount_cert_3, CERT_FEE)
+            assert (len(cert_3_epoch_0) > 0)
+            mark_logs("Certificate is {}".format(cert_3_epoch_0), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send certificate failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert (False)
+
+        # Checking all certificate in mempool
+        self.sync_all()
+        assert_equal(True, cert_3_epoch_0 in self.nodes[0].getrawmempool())
+
+        mined = self.nodes[0].generate(1)[0]
+        self.sync_all()
+
+        # Check balance
+        bwt_address = self.nodes[1].getrawcertificate(cert_3_epoch_0, 1)['vout'][1]['scriptPubKey']['addresses'][0]
+        res = self.nodes[1].listtxesbyaddress(bwt_address)
+        assert_equal(res[0]['scid'], scid)
+        mark_logs("Height : {}".format(res[0]['vout'][1]['maturityHeight']), self.nodes, DEBUG_MODE)
+        ud = self.nodes[1].getunconfirmedtxdata(bwt_address)
+        assert_equal(bwt_amount_3, ud['bwtImmatureOutput'])
+        mark_logs("Immature output : {}".format(ud['bwtImmatureOutput']), self.nodes, DEBUG_MODE)
+
+        # Checking all certificate in mempool
+        mark_logs("Check certificates is not in mempool anymore", self.nodes, DEBUG_MODE)
+        assert_equal(False, cert_1_epoch_0 in self.nodes[0].getrawmempool())
+        assert_equal(False, cert_2_epoch_0 in self.nodes[0].getrawmempool())
+        assert_equal(False, cert_3_epoch_0 in self.nodes[0].getrawmempool())
+
+        height = self.nodes[0].getblockcount()
+        mark_logs("Check certificates in chain", self.nodes, DEBUG_MODE)
+        assert_true(cert_1_epoch_0 in self.nodes[0].getblock(self.nodes[0].getblockhash(height - 1), True)['cert'])
+        assert_true(cert_2_epoch_0 in self.nodes[0].getblock(self.nodes[0].getblockhash(height - 1), True)['cert'])
+        assert_true(cert_3_epoch_0 in self.nodes[0].getblock(mined, True)['cert'])
+
+        mark_logs("Checking certificates in mempool after block revertion(via invalidateblock)", self.nodes, DEBUG_MODE)
+        block_inv = self.nodes[0].getbestblockhash()
+        self.nodes[0].invalidateblock(block_inv)
```

Comment:
It would be more interesting to have node 1 invalidating the block (see also comment below) because the bwts of the certs have been sent to its wallet